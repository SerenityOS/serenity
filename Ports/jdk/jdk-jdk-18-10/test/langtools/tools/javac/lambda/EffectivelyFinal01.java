/*
 * @test /nodynamiccopyright/
 * @bug 8003280
 * @summary Add lambda tests
 *  effectively final check fails on method parameter
 * @compile/fail/ref=EffectivelyFinal01.out -XDrawDiagnostics EffectivelyFinal01.java
 */
class EffectivelyFinal01 {

    interface SAM {
        Integer m(Integer i);
    }

    void test(Integer nefPar) {
        SAM s = (Integer h) ->  { Integer k = 0; return k + h + nefPar; };
        nefPar++;  //non-effectively final
    }
}
