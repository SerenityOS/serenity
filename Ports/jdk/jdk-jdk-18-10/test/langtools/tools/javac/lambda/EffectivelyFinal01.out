EffectivelyFinal01.java:15:65: compiler.err.cant.ref.non.effectively.final.var: nefPar, (compiler.misc.lambda)
1 error
